t/prometheus-metrics.t

```diff
@@ -275,7 +275,137 @@ qr/upstream_response_time_seconds(.|\n)*upstream_response_time_seconds_bucket\{l
 --- response_body_like eval
 [
 "",
-qr/total_response_time_seconds(.|\n)*total_response_time_seconds_bucket\{le=".*"\} 1/
+qr/total_response_time_seconds(.|\n)*total_response_time_seconds_bucket\{service="all",le=".*"\} 1/
+]
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: extended metrics show services id
+--- env eval
+("APICAST_EXTENDED_METRICS", "true")
+--- configuration
+{
+  "services": [
+    {
+      "id": 42,
+      "proxy": {
+        "policy_chain": [
+          {
+            "name": "apicast.policy.upstream",
+            "configuration": {
+              "rules": [
+                {
+                  "regex": "/",
+                  "url": "http://test:$TEST_NGINX_SERVER_PORT"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+--- upstream
+  location / {
+     content_by_lua_block {
+       ngx.exit(200);
+     }
+  }
+
+  location /two {
+     content_by_lua_block {
+       ngx.exit(200);
+     }
+  }
+
+--- request eval
+["GET /", "GET /metrics"]
+--- more_headers eval
+["", "Host: metrics"]
+--- error_code eval
+[ 200, 200 ]
+--- response_body_like eval
+[
+"",
+qr/total_response_time_seconds(.|\n)*total_response_time_seconds_bucket\{service="42",le=".*"\} 1/,
+qr/upstream_response_time_seconds(.|\n)*upstream_response_time_seconds_bucket\{service="42",le=".*"\} 1(.|\n)*upstream_status\{status="42",service="all"\} 1/
+]
+--- no_error_log
+[error]
+
+
+=== TEST 7: extended metrics with multiple services
+--- env eval
+("APICAST_EXTENDED_METRICS", "true")
+--- configuration
+{
+  "services": [
+    {
+      "id": 42,
+      "proxy": {
+        "policy_chain": [
+          {
+            "name": "apicast.policy.upstream",
+            "configuration": {
+              "rules": [
+                {
+                  "regex": "/",
+                  "url": "http://test:$TEST_NGINX_SERVER_PORT"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    },
+    {
+      "id": 21,
+      "proxy": {
+        "policy_chain": [
+          {
+            "name": "apicast.policy.upstream",
+            "configuration": {
+              "rules": [
+                {
+                  "regex": "/",
+                  "url": "http://test:$TEST_NGINX_SERVER_PORT/two"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+--- upstream
+  location / {
+     content_by_lua_block {
+       ngx.exit(200);
+     }
+  }
+
+  location /two {
+     content_by_lua_block {
+       ngx.exit(200);
+     }
+  }
+--- request eval
+["GET /", "GET /metrics"]
+--- more_headers eval
+["", "Host: metrics"]
+--- error_code eval
+[ 200, 200 ]
+--- response_body_like eval
+[
+"",
+qr/total_response_time_seconds(.|\n)*total_response_time_seconds_bucket\{service="42",le=".*"\} 1/,
```

Comment:
If we do more of these we could refactor this into some helper method.

So we could say just `$prometheus->match_metrics('total_response_time_seconds', 'total_response_time_seconds_bucket')` and that helper method would generate the regexp.

Reply:
I would like to defer this one for a while, my head is already exploding with Lua :-D

Reply:
I can imagine :-D For the record, there are a few helpers already:
https://github.com/3scale/APIcast/blob/master/t/policies.pl
https://github.com/3scale/APIcast/blob/master/t/fixtures/mutual_ssl.pl
https://github.com/3scale/APIcast/blob/master/t/fixtures/tls.pl